pkg/kubecost/window.go

```diff
@@ -432,6 +447,86 @@ func (w Window) Minutes() float64 {
 	return w.end.Sub(*w.start).Minutes()
 }
 
+// Overlaps returns true iff the two given Windows share and amount of temporal
+// coverage.
+// TODO complete (with unit tests!) and then implement in AllocationSet.accumulate
+// TODO:CLEANUP
+func (w Window) Overlaps(x Window) bool {
+	if (w.start == nil && w.end == nil) || (x.start == nil && x.end == nil) {
+		// one window is completely open, so overlap is guaranteed
+		// <---------->
+		//   ?------?
+		return true
+	}
+
+	// Neither window is completely open (nil, nil), but one or the other might
+	// still be future- or past-open.
+
+	if w.start == nil {
+		// w is past-open, future-closed
+		// <------]
+
+		if x.start != nil && !x.start.Before(*w.end) {
```

Comment:
`!x.start.Before(*w.end)` vs. `x.start.After(*w.end)`?

Reply:
I guess this also introduces the question of if `Before` and `After` are inclusive (i.e. `<=` `>=`) or not and therefore also the question of "if they share a single instant (w.end = x.start) do they overlap?"

Reply:
"Not before" is the same as "After or equal to", so inclusivity is the difference. You question regarding the definition of "overlap" is a good one. Going to defer that decision until we implement the function. (n the meantime, I've commented it all out, but left the WIP and the TODOs.)